battle-engine.js

```diff
@@ -2043,7 +2043,7 @@ Battle = (function () {
 			this.debug(eventid + ' handler suppressed by Gastro Acid, Klutz or Magic Room');
 			return relayVar;
 		}
-		if (target.ignore && target.ignore[effect.effectType + 'Target']) {
+		if (effectType === 'Weather' && eventid !== 'TryWeather' && !this.runEvent('TryWeather', target)) {
```

Comment:
`line 2046  col 13  'effectType' is not defined.`

It should be `effect.effectType`, I think. Also, make sure you test the code yourself before you push.